pkg/dsl/definition/package.go

```diff
@@ -0,0 +1,187 @@
+package definition
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/ast"
+	"cuelang.org/go/cue/build"
+	"cuelang.org/go/cue/token"
+	"cuelang.org/go/encoding/jsonschema"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
+)
+
+var velaBuiltinPkgs []*build.Instance
+
+func addImportsFor(bi *build.Instance) {
+	bi.Imports = append(bi.Imports, velaBuiltinPkgs...)
+}
+
+// AddImportFromCluster use  K8s native API and CRD definition as a reference package in template rendering
+func AddImportFromCluster(config *rest.Config, scheme *runtime.Scheme) error {
+	copyConfig := *config
+	apiSchema, err := getClusterOpenAPI(&copyConfig, scheme)
+	if err != nil {
+		return err
+	}
+	kubePkg := newPackage("kube")
+	if err := kubePkg.addOpenAPI(apiSchema); err != nil {
+		return err
+	}
+	kubePkg.mount()
+	return nil
+}
+
+func getClusterOpenAPI(config *rest.Config, scheme *runtime.Scheme) (string, error) {
+	codec := runtime.NoopEncoder{Decoder: serializer.NewCodecFactory(scheme).UniversalDeserializer()}
```

Comment:
Why create this codec?

Reply:
restConfig's NegotiatedSerializer needs to be initialized to generate UnversionedRESTClient, but codec is not necessary